### PR TITLE
Ft/s3 c 1146 get metadata blob

### PR DESCRIPTION
**Pull request by rahulreddy**
- fix: update mem versioning to reflect other backends
- ft: backbeat get metadata blob route

### CODE REVIEW

File: lib/metadata/in_memory/backend.js [hunk 1]
```diff
@@ -83,6 +83,27 @@ const metastore = {
...
+                    const mst = metadata.keyMaps.get(bucketName).get(objName);
+                    if (mst && mst.versionId === params.versionId || !mst) {
+                        metadata.keyMaps.get(bucketName).set(objName, objVal);
+                    }
+                    // eslint-disable-next-line
+                    objName = formatVersionKey(objName, params.versionId);
+                    metadata.keyMaps.get(bucketName).set(objName, objVal);
+                    return cb(null, `{"versionId":"${objVal.versionId}"}`);
+                }
                 if (params && params.versioning) {
```
Comment (philipyoo) [inline lib/metadata/in_memory/backend.js]: Just checking, should `params.versionId === ''` be moved above this check?  if `params.versionId === ''` and `params.versioning`

Comment (rahulreddy) [reply lib/metadata/in_memory/backend.js]: If versioning is true, then we expect either no version specified (undefined) or a valid versionId. The next one is for versioning suspended buckets I believe. It should be ok for that case.

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
@ironman-machine try

**Comment by rahulreddy** [comment]
Integration passed here on retry http://ci.ironmann.io/gh/scality/Integration/18612

**Comment by bennettbuchanan** [comment]
@ironman-machine 
SCALITY_BACKBEAT_BRANCH=ft/S3C-1146-getandputMetadata
try

**Comment by bennettbuchanan** [comment]
Updated backbeat to reuse the datastoreVersionId from external backend, rerunning E2E.

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_BACKBEAT_BRANCH=ft/S3C-1146-getandputMetadata
try

**Pull request merged**